pandas/tests/util/test_assert_series_equal.py

```diff
@@ -194,3 +194,20 @@ def test_series_equal_categorical_mismatch(check_categorical):
             tm.assert_series_equal(s1, s2, check_categorical=check_categorical)
     else:
         _assert_series_equal_both(s1, s2, check_categorical=check_categorical)
+
+
+def test_series_equal_series_type():
+    class MySeries(Series):
+        pass
+
+    s1 = Series([1, 2])
+    s2 = Series([1, 2])
+    s3 = MySeries([1, 2])
+
+    tm.assert_series_equal(s1, s2, check_series_type=False)
+    tm.assert_series_equal(s1, s2, check_series_type=True)
+
+    tm.assert_series_equal(s1, s3, check_series_type=False)
+
+    with pytest.raises(AssertionError, match="Series type not equal"):
+        tm.assert_series_equal(s1, s3, check_series_type=True)
```

Comment:
should this test include tm.assert_series_equal(s3, s1, check_series_type=True) (with an xfail))

Reply:
It should but the original `check_series_type` should also be extended as
`isinstance(s3, type(s1))` is True while `isinstance(s1, type(s3))` is False. `assert_series_equal` should probably return the same result no matter the order of left and right Series.

Reply:
yeah, I think the fix is outside of the scope of this PR since this is simply reverting changes. but if we are adding a new test, the test should probably test all permutations for completeness. Maybe add a TODO comment in the test instead.

Reply:
I think we can just fix it here at once?

Reply:
could maybe remove the comments in assert_series_equal now?

Reply:
> I think we can just fix it here at once?

@jorisvandenbossche that broke (some) tests.

Reply:
Ah, should be easy to fix the test to make proper use of assert_frame_equal, I think (will look at it tomorrow)